### ISSUE TITLE
home bridge unable to connect after setup with HomeKit.

### ISSUE THREAD

**Issue by damdude** [haywirecoder/homebridge-envisalink-ademco#20]
[config.txt](https://github.com/haywirecoder/homebridge-envisalink-ademco/files/8011811/config.txt)
[homebridge-error.log](https://github.com/haywirecoder/homebridge-envisalink-ademco/files/8011813/homebridge-error.log)
[homebridge-out.log](https://github.com/haywirecoder/homebridge-envisalink-ademco/files/8011814/homebridge-out.log)

I am running into this issue which I am not sure where is it coming from.  I am running homebridge on TrueNas (version 12.0) as plugin. Homebridge version is 1.4.0. I have Honeywell Vista 20p board and its hooked up to Envisalink EVL4 which works fine. Looks like home bridge plugin for Envisalink connects fine with Envisalink board and get its heartbeat fine (confirmed from logs) but the moment I setup this device with apple Home app (setting up Envisalink home bridge as child process, so that Envisalink shows up directly) the home bridge goes out in bad state and doesn't give me access back to its dashboard. I am attaching my config and logs from home bridge. I have taken out the sensitive data and put like this "<<SENSITIVE_INFO >>>". From logs it sounds like the Homebrige is not able to make the connection anymore. any kind of help will be greatly appreciated.

**Comment by haywirecoder** [comment]
Hi,

I haven't tried running this plug-in as a child Homebridge, so I can't speak directly to your configuration. Does the plug-in work correctly when initially setup as part of main Homebridge instance? The error you're receiving is a low level error on network stack (e.g. your losing network connection). Do you have other plug-in running that has successful connected to the Home app?

**Comment by damdude** [comment]
> Hi,
> 
> 
> 
> I haven't tried running this plug-in as a child Homebridge, so I can't speak directly to your configuration. Does the plug-in work correctly when initially setup as part of main Homebridge instance? The error you're receiving is a low level error on network stack (e.g. your losing network connection). Do you have other plug-in running that has successful connected to the Home app?
> 
> 

I am not sure if plugin works fine if I don't set it up as child. I'll need to reinstall the homebridge to confirm that. But I know for sure everything looks fine until I try to complete setup using Apple home app. I also noticed that plug-in is able to get the heartbeat n system status until I start setting it up with Apple home app. I haven't setup any other plug-in yet so don't know what else could be causing this issue. I actually installed homebridge just for this envisalink only. I have read in other forums that envisalink doesn't support multiple network connections to its board. But there is no other connection. I wonder me setting up online account with envisalink website is causing this coz whenever I arm or disarm system I get text sms from the eyezon.com

**Comment by haywirecoder** [comment]
Hi,

If you only have one plug-in not clear why your creating a child bridge, unless your planning to install a lot of other plug-ins. My suggestion, setup clean version of Homebridge without any plug-in and try connecting to Home App. Then install a simple switch using homebridge-dummy and make sure the connection between Home and Homebridge is functional. If Homebridge itself is crashing thats indication something is incorrect in your install. I would debug that issue before installing the plug-in. 

You are correct only one local connection to EVL4 is possible, the online account however does not take this connections. You can turn off notification using Online EZ portal if you wish.

**Comment by damdude** [comment]
Okay sounds good. I will do a fresh homebridge install and will try to check if empty homebridge itself works fine with home app or not.

**Comment by damdude** [comment]
Accidently closed it. I would like to keep it open until I come back with results of last experiment suggested.

**Comment by damdude** [comment]
You are right, the fresh install with no plugin also shows the same issue. but I am not sure what is happening. Log file doesn't  have error listed in it. I wonder if my system is not strong enough, could that be a issue ?

Below is the version installed with the plugin, 
2022-02-09T19:07:12: PM2 log: PM2 version          : 5.1.2
2022-02-09T19:07:12: PM2 log: Node.js version     : 16.13.0

**Comment by damdude** [comment]
Okay, I found out what's the issue is. I guess sometimes having a complex and too many IOT device can also counter work.  It turns out  when for the first time I setup my homebridge with My apple home App, by default the security of this bridge is set to "Restricted". ( I have a HomeKit compatible router-EERO,which gives me the capability to restrict the homekit devices to access internet and only talk to the local devices.). so the moment I setup my homebridge with apple home app, I guess homebridge is trying to access outside local network and my router didn't allowed it, it goes into this constant processing mode and hence never comes back from it. So the moment, I changed the restriction from "Restricted" to "Auto" ( meaning, it can talk to outside my network but only to approved list of manufacturers) things connected well and homebridge status came back normal and I was able to setup this envisalink plugin. Now, right after I setup this plugin, I noticed few things which shouldn't and should happen for which I guess I will open a new issue and enhancement request. Also, I am wondering this wiki can be used for getting some more infor in terms of how to bring an offline fire sensor connected to the system back online including wiring and system programming.
